src/interfaces/SpokePool.ts

```diff
@@ -230,8 +230,7 @@ export interface v3RelayData extends RelayDataCommon {
   exclusivityDeadline: number;
 }
 
-// @todo: Extend with v2RelayData | v3RelayData.
-export type RelayData = v2RelayData;
+export type RelayData = v2RelayData | v3RelayData;
```

Comment:
Should we merge the v3RelayData and Fill objects somehow? At the contract level, the following functions all take in a `V3RelayData` object:
- `fillV3Relay`
- `fillV3RelayWithUpdatedDeposit`
- `requestV3SlowFill`
- `executeV3SlowRelayLeaf`
- `_getV3RelayHash`

Reply:
Yep, I like this. Have done some initial work towards it here: https://github.com/across-protocol/sdk-v2/commit/e7d67a50175884f70fe3d98cd7caf7d6457cd6c6